geode-core/src/main/java/org/apache/geode/internal/cache/PRHARedundancyProvider.java

```diff
@@ -1981,13 +1988,12 @@ public InternalPartitionDetails buildPartitionMemberDetails(final boolean intern
    * whichever happens first.
    */
   protected void waitForPersistentBucketRecoveryOrClose() {
-    CountDownLatch recoveryLatch = allBucketsRecoveredFromDisk;
-    if (recoveryLatch != null) {
+    if (getPersistentBucketRecoverer() != null) {
       boolean interrupted = false;
       while (true) {
         try {
           this.prRegion.getCancelCriterion().checkCancelInProgress(null);
-          boolean done = recoveryLatch.await(
+          boolean done = getPersistentBucketRecoverer().await(
```

Comment:
I think this whole while loop should be in the await method on the PersistentBucketRecoverer. We passed "this" into the constructor so it should know about "this.prRegion" so that it can check the cancel criterion.